src/typedefs.rs

```diff
@@ -395,6 +395,30 @@ macro_rules! construct_secret_key {
         /// # Security:
         /// - __**Avoid using**__ `unprotected_as_bytes()` whenever possible, as it breaks all protections
         /// that the type implements.
+        ///
+        /// - The trait `PartialEq<&'_ [u8]>` is implemented for this type so that users are not tempted
+        /// to call `unprotected_as_bytes` to compare this sensitive value to a byte string. The trait
+        /// is implemented in such a way that the comparison happens in constant time. Thus, users should
+        /// prefer `SecretType == &[u8]` over `SecretType.unprotected_as_bytes() == &[u8]`.
+        /// Examples are shown below. Although the `orion::hazardous::stream::chacha20::SecretKey`
+        /// type is used, this example applies to any fixed-size store of sensitive data.
```

Comment:
It also applies to the variable-sized (heap) secret values, not only fixed-size. Generally I think this is clear from any type that implements said trait, so maybe we can get rid of "Although the [..] store of sensitive data." altogether to also keep the bullet-point a bit more concise?

Reply:
Or say "The examples apply to any type that implements `PartialEq<&'_ [u8]>`. That would be more clear.

Reply:
I like the phrasing in your second example. "The examples apply to any type that implements `PartialEq<&'_ [u8]>`". Will make the change.